src/NorthstarClient.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace DoSomething\Northstar;
+
+use DoSomething\Northstar\Common\RestAPIClient;
+use DoSomething\Northstar\Exceptions\UnauthorizedException;
+use DoSomething\Northstar\Exceptions\ValidationException;
+use DoSomething\Northstar\Resources\NorthstarKey;
+use DoSomething\Northstar\Resources\NorthstarUser;
+use DoSomething\Northstar\Resources\NorthstarUserCollection;
+use DoSomething\Northstar\Resources\NorthstarKeyCollection;
+
+class NorthstarClient extends RestAPIClient
+{
+    /**
+     * Create a new Northstar API client.
+     * @param array $config
+     */
+    public function __construct($config = [])
+    {
+        $base_url = $config['url'].'/v1/';
+        $api_key = $config['api_key'];
+
+        parent::__construct($base_url, ['X-DS-REST-API-Key' => $api_key]);
+    }
+
+    /**
+     * Send a POST request to verify the user's credentials.
+     *
+     * @param array $credentials
+     *   ex: ['email' => '...', 'password' => '...']
+     *       ['mobile' => '...', 'password' => '...']
+     * @return NorthstarUser|null
+     */
+    public function verify($credentials)
+    {
+        try {
+            $response = $this->post('auth/verify', $credentials);
+
+            return new NorthstarUser($response['data']);
+        } catch (UnauthorizedException $e) {
+            return null;
+        } catch (ValidationException $e) {
+            return null;
```

Comment:
I personally like `return null;` to be explicit about what's going on (since `return;` requires a bit more thinking about how PHP handles a function that doesn't return a value). Happy to change though.